resource/profile.rs

```diff
@@ -0,0 +1,767 @@
+                    - data.snapshot_commit_record_create_nanos
+                    - data.snapshot_durable_write_data_submit_nanos
+                    - data.snapshot_isolation_validate_nanos
+            },
+        }
+    }
+
+    pub fn snapshot_storage_written(&mut self) {
+        match &mut self.data {
+            None => {}
+            Some(data) => {
+                data.snapshot_storage_write_nanos = Instant::now().duration_since(data.start).as_nanos()
+                    - data.types_validation_nanos
+                    - data.things_finalise_nanos
+                    - data.functions_finalise_nanos
+                    - data.schema_update_statistics_durable_write_nanos
+                    - data.snapshot_put_statuses_check_nanos
+                    - data.snapshot_commit_record_create_nanos
+                    - data.snapshot_durable_write_data_submit_nanos
+                    - data.snapshot_isolation_validate_nanos
+                    - data.snapshot_durable_write_data_confirm_nanos
+            },
+        }
+    }
+
+    pub fn snapshot_isolation_manager_notified(&mut self) {
+        match &mut self.data {
+            None => {}
+            Some(data) => {
+                data.snapshot_isolation_manager_notify_nanos = Instant::now().duration_since(data.start).as_nanos()
+                    - data.types_validation_nanos
+                    - data.things_finalise_nanos
+                    - data.functions_finalise_nanos
+                    - data.schema_update_statistics_durable_write_nanos
+                    - data.snapshot_put_statuses_check_nanos
+                    - data.snapshot_commit_record_create_nanos
+                    - data.snapshot_durable_write_data_submit_nanos
+                    - data.snapshot_isolation_validate_nanos
+                    - data.snapshot_durable_write_data_confirm_nanos
+                    - data.snapshot_storage_write_nanos
+            },
+        }
+    }
+
+    pub fn snapshot_durable_write_commit_status_submitted(&mut self) {
+        match &mut self.data {
+            None => {}
+            Some(data) => {
+                data.snapshot_durable_write_commit_status_submit_nanos = Instant::now().duration_since(data.start).as_nanos()
+                    - data.types_validation_nanos
+                    - data.things_finalise_nanos
+                    - data.functions_finalise_nanos
+                    - data.schema_update_statistics_durable_write_nanos
+                    - data.snapshot_put_statuses_check_nanos
+                    - data.snapshot_commit_record_create_nanos
+                    - data.snapshot_durable_write_data_submit_nanos
+                    - data.snapshot_isolation_validate_nanos
+                    - data.snapshot_durable_write_data_confirm_nanos
+                    - data.snapshot_storage_write_nanos
+                    - data.snapshot_isolation_manager_notify_nanos
+            },
+        }
+    }
```

Comment:
This is a very unsophisticated approach, and will break immediately if we change the order of commits. If we can find something as simply/fast/roboust to skipping individual steps, that would work for me too!